### PR TITLE
upcoming: [DI-22390] - Added disabled dashboard select in contextual view

### PR DESCRIPTION
**Pull request by nikhagra-akamai**
## Description 📝

Highlight the Pull Request's context and intentions.

## Changes  🔄

List any change(s) relevant to the reviewer.

- ...
- ...

## Target release date 🗓️

Please specify a release date (and environment, if applicable) to guarantee timely review of this PR. If exact date is not known, please approximate and update it as needed.

## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

Screenshot 2024-12-13 at 1 36 58 PM

### Prerequisites

(How to setup test environment)

- ...
- ...

### Reproduction steps

(How to reproduce the issue, if applicable)

- [ ] ...
- [ ] ...

### Verification steps

(How to verify changes)

- [ ] ...
- [ ] ...

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

*Check all that apply*

- [ ] Use React components instead of HTML Tags
- [ ] Proper naming conventions like cameCase for variables & Function & snake_case for constants
- [ ] Use appropriate types & avoid using "any"
- [ ] No type casting & non-null assertions
- [ ] Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] Providing/Improving test coverage
- [ ] Use sx props to pass styles instead of style prop
- [ ] Add JSDoc comments for interface properties & functions
- [ ] Use strict equality (===) instead of double equal (==)
- [ ] Use of named arguments (interfaces) if function argument list exceeds size 2
- [ ] Destructure the props
- [ ] Keep component size small & move big computing functions to separate utility
- [ ] 📱 Providing mobile support

<br/>

- [ ] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [ ] All unit tests are passing
- [ ] TypeScript compilation succeeded without errors
- [ ] Code passes all linting rules

</details>

---

## Commit message and pull request title format standards

> **Note**: Remove this section before opening the pull request
**Make sure your PR title and commit message on squash and merge are as shown below**

`<commit type>: [JIRA-ticket-number] - <description>`

**Commit Types:**

- `feat`: New feature for the user (not a part of the code, or ci, ...).
- `fix`: Bugfix for the user (not a fix to build something, ...).
- `change`: Modifying an existing visual UI instance. Such as a component or a feature.
- `refactor`: Restructuring existing code without changing its external behavior or visual UI. Typically to improve readability, maintainability, and performance.
- `test`: New tests or changes to existing tests. Does not change the production code.
- `upcoming`: A new feature that is in progress, not visible to users yet, and usually behind a feature flag.

**Example:** `feat: [M3-1234] - Allow user to view their login history`

---

### CODE REVIEW

File: packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx [hunk 1]
```diff
@@ -10,16 +10,24 @@ import type { Dashboard, FilterValue } from '@linode/api-v4';
...
 export interface CloudPulseDashboardSelectProps {
   defaultValue?: Partial<FilterValue>;
-  handleDashboardChange: (
+  disabled?: boolean;
+  handleDashboardChange?: (
     dashboard: Dashboard | undefined,
     savePref?: boolean
   ) => void;
   savePreferences?: boolean;
+  serviceIntegration?: boolean;
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx]: I think we can just control everything based on serviceIntegration Prop and no need of disabled prop itself

File: packages/manager/src/features/CloudPulse/Dashboard/CloudPulseDashboardWithFilters.tsx [hunk 2]
```diff
@@ -118,28 +119,52 @@ export const CloudPulseDashboardWithFilters = React.memo(
...
+                rowSpacing={2}
+              >
+                <Grid display={'flex'} item md={4} sm={5} xs={12}>
+                  <CloudPulseDashboardSelect
+                    defaultValue={dashboardId}
+                    disabled
+                    serviceIntegration
+                  />
+                </Grid>
+                <Grid display="flex" gap={1} item md={4} sm={5} xs={12}>
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/Dashboard/CloudPulseDashboardWithFilters.tsx:143]: do we need gap here, just asking

File: packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx [hunk 3]
```diff
@@ -10,16 +10,24 @@ import type { Dashboard, FilterValue } from '@linode/api-v4';
 
 export interface CloudPulseDashboardSelectProps {
   defaultValue?: Partial<FilterValue>;
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx:15]: General comments, here we missed to explain what these props are used for, Can we include that as well

File: packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx [hunk 4]
```diff
@@ -103,7 +111,7 @@ export const CloudPulseDashboardSelect = React.memo(
         autoHighlight
         clearOnBlur
         data-testid="cloudpulse-dashboard-select"
-        disabled={!dashboardsList}
+        disabled={disabled || !dashboardsList}
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx]:
```suggestion
        disabled={isServiceIntegration || !dashboardsList}
```

File: packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx [hunk 5]
```diff
@@ -10,16 +10,24 @@ import type { Dashboard, FilterValue } from '@linode/api-v4';
...
 
 export const CloudPulseDashboardSelect = React.memo(
   (props: CloudPulseDashboardSelectProps) => {
-    const { defaultValue, handleDashboardChange, savePreferences } = props;
+    const {
+      defaultValue,
+      disabled,
+      handleDashboardChange = () => {},
+      savePreferences,
+      serviceIntegration: isServiceIntegration,
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardSelect.tsx]: we can rename the original prop itself as **isServiceIntegration**

**Comment by venkymano-akamai** [comment]
@nikhagra-akamai , Can you just add one screenshot of how it looks like

**Comment by nikhagra-akamai** [comment]
> @nikhagra-akamai , Can you just add one screenshot of how it looks like

I've added in the description

File: packages/manager/src/features/CloudPulse/Dashboard/CloudPulseDashboardWithFilters.tsx (hunk 2 above)
Comment (nikhagra-akamai) [reply packages/manager/src/features/CloudPulse/Dashboard/CloudPulseDashboardWithFilters.tsx:143]: I've taken this complete structure from global filters. I've to see if it is required and need to modify in global filters as well

**Comment by nikhagra-akamai** [comment]
Screenshot 2024-12-13 at 1 36 58 PM

**Comment by nikhagra-akamai** [comment]
Screenshot 2024-12-13 at 1 51 53 PM

**Review by venkymano-akamai — APPROVED** [review]

**Pull request merged**